tiatoolbox/models/architecture/unet.py

```diff
@@ -220,20 +226,33 @@ def __init__(
         num_input_channels: int = 2,
         num_output_channels: int = 2,
         encoder: str = "resnet50",
-        decoder_block: Tuple[int] = (3, 3),
+        encoder_levels: List[int] = None,
+        decoder_block: Tuple[int] = None,
+        skip_type: str = "add",
```

Comment:
Intial design was to hard lock only 2 form of encoders. Now it becomes quite counter intuitive between `encoder_levels` and `encoder`. Should retain hard lock and expose overide for `encoder` (as in receiving a model).

Reply:
encoder_levels has been incorporated to support for the flexibility of Unet encoder choice. This is only applicable for UNet encoder as described in the docstrings